### ISSUE TITLE
RFE: add support for `else if` operator

### ISSUE THREAD

**Issue by ignatenkobrain** [rpm-software-management/rpm#311]
having things like

```rpmspec
%if 0%{?fedora} == 26
...
%else
%if 0%{?fedora} > 26
...
%else
%if 0%{?rhel}
...
%endif
%endif
%endif
```

does the job, but completely unreadable. I don't think it would be hard to implement `%else if` or `%elif`, but would be very useful.

**Comment by n3npq** [comment]
Yes %if tests on values are entirely unreadable, and adding additional constructs like %elif just encourages the madness.

There are many ways to avoid using %if constructions entirely by writing more intelligent macros, or providing a better feature based (rather than arch/osversion value based) logic.

**Comment by n3npq** [comment]
For example:

Once can arrange for case-like set of inclusions by adding the value of %fedora to a filename and adding the snippets as SourceN: directives that are then selectively as a file path by an appropriate expansion.

All I'm saying here is adding snippets as fedora_%{?fedora} files that can then be selected in a single line in a spec file like
```
    %include %{_sourcedir}/fedora_%{?fedora}
```

There are lots of other simplifications that are far more readable than %if syntax if you think a bit.

**Comment by Qantas94Heavy** [comment]
Wouldn't that require you to create a specific include file (which could duplicate a lot of stuff) for each version? You might also need another conditional to handle RHEL for the OP's sample case.

I can't see how that is much better than using if-else condtions -- both have their problems.

**Comment by Conan-Kudo** [comment]
Regardless, having an `%elif` conditional statement makes sense. It's also a bit weird that we haven't had it for so long...

**Comment by n3npq** [comment]
The difference is that %elif needs to be manually replicated in many spec files, while a common stanza instantiated in a file can be included where needed.

IOW refactoring a structural solution to some configuration problem is far easier.

The logic is (of course) identical.

**Comment by pmatilai** [comment]
Implemented as of commit 1c4b238840d0995344d0d0381a0561b213429203
